jdaviz/configs/mosviz/plugins/parsers.py

```diff
@@ -532,27 +502,15 @@ def mos_niriss_parser(app, data_dir, obs_label=""):
 
                         data = temp[sci].data
                         meta = temp[sci].header
-                        header = filter_wcs[filter_name]
-
-                        # Information that needs to be added to the header in order to load
-                        # WCS information into SpectralCube.
-                        # TODO: Use gwcs instead to avoid hardcoding information for 3rd wcs axis
-                        new_data = np.expand_dims(data, axis=1)
-                        header['NAXIS'] = 3
-
-                        header['NAXIS3'] = 1
-                        header['BUNIT'] = 'dN/s'
-                        header['CUNIT3'] = 'um'
-                        header['WCSAXES'] = 3
-                        header['CRPIX3'] = 0.0
-                        header['CDELT3'] = 1E-06
-                        header['CUNIT3'] = 'm'
-                        header['CTYPE3'] = 'WAVE'
-                        header['CRVAL3'] = 0.0
-
-                        wcs = WCS(header)
-
-                        spec2d = SpectralCube(new_data, wcs=wcs, meta=meta)
+
+                        wcs = WCS(naxis=1)
+                        wcs.wcs.ctype = ['WAVE']
+                        wcs.wcs.crpix = [0]
+                        wcs.wcs.crval = [0]
+                        wcs.wcs.cdelt = [1.e-6]
+                        wcs.wcs.cunit = ['m']
```

Comment:
Note that this is still a fake spectral WCS, as it was before.

Reply:
Is there a reason why we still need to fake the WCS @astrofrog? Particularly with #735's strategy to link by pixel, I imagine a WCS, even for Mosviz, is no longer an absolute requirement? I think we were hoping that with transitioning to Spectrum1D, we could drop these fake values?

My understanding was that the WCS was required to construct a SpectralCube, hence why faking these values were required. I don't believe Spectrum1Ds have this requirement?

Reply:
Yes we could just not add a WCS and link by pixels in this case.

Reply:
It sounds in offband conversations, the Product Owners are favoring just pushing this through. It seems Pixel Linking will only stall this PR further, so we can tackle this improvement at a later date